src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,62 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private int resultSupply = 0;
+    private int resultBuy = 0;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        List<String> lines = readFromFile(fromFileName);
+
+        for (String dataElement : lines) {
+            String[] arrayDataSplit = dataElement.split(",");
+
+            if (arrayDataSplit.length == 2) {
+                String action = arrayDataSplit[0].trim();
+                int amount = Integer.parseInt(arrayDataSplit[1].trim());
+                if (action.equals(SUPPLY)) {
+                    resultSupply += amount;
+                } else if (action.equals(BUY)) {
+                    resultBuy += amount;
+                }
+            }
+        }
+
+        String report = createReport();
+
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(toFileName))) {
+            bufferedWriter.write(report);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't write data to " + toFileName, e);
+        }
+    }
```

Comment:
Please split logic into 3 private method and leave in public method only calls of these 3 methods